### PR TITLE
CERD-205 > Drag and drop fixes/enhancements

### PR DESCRIPTION
**Pull request by drochgenius**
Hi @adjamenhmh , essentially, I have implemented support for nested drag and drop containers.

### CODE REVIEW

File: packages/drag-drop/src/unit/rendering.spec.ts [hunk 1]
```diff
@@ -172,6 +144,29 @@ export default () => {
...
+            withSnippet('nested-containers');
+            const element: DragDrop = document.querySelector(basicTagName) as any;
+            await element.updateComplete;
+
+            // @ts-ignore Access private member
+            expect(element.dragContainers.length).to.equal(1);
+            // @ts-ignore Access private member
+            expect(element.dropContainers.length).to.equal(3);
+        });
+
```
Comment (drochgenius) [inline packages/drag-drop/src/unit/rendering.spec.ts]: Added these tests @adjamenhmh

File: packages/drag-drop/test/runner.html [hunk 2]
```diff
@@ -150,6 +153,34 @@ <h3>Sort These Items</h3>
...
+            <section id="draggable-area">
+                <drag-container>
+                    <div slot="options" id="monkey">Monkey</div>
+                    <div slot="options" id="frog">Frog</div>
+                    <div slot="options" id="mouse">Mouse</div>
+                    <div slot="options" id="rabbit">Mouse</div>
+                </drag-container>
+            </section>
+        </drag-drop>
+    </template>
```
Comment (drochgenius) [inline packages/drag-drop/test/runner.html]: + this template in the runner

File: packages/drag-drop/src/components/drag-drop.ts [hunk 3]
```diff
@@ -136,15 +136,28 @@ export class DragDrop extends ComponentBase<string[]> {
...
+            this.dropContainers.push(el as DropContainer);
+        } else if (typeof el.querySelectorAll === 'function') {
+            el.querySelectorAll('drop-container').forEach(
+                (container: HTMLElement): void => {
+                    this.dropContainers.push(container as DropContainer);
+                }
+            );
+            el.querySelectorAll('drag-container').forEach(
+                (container: HTMLElement): void => {
+                    this.dragContainers.push(container as DragContainer);
```
Comment (drochgenius) [inline packages/drag-drop/src/components/drag-drop.ts]: + this is the most relevant change to support nested containers for both drag and drop.

Comment (adjamenhmh) [reply packages/drag-drop/src/components/drag-drop.ts]: @drochgenius  I'm wondering if we should support similar cases for drag-container and their option items

Comment (drochgenius) [reply packages/drag-drop/src/components/drag-drop.ts]: no

**Pull request merged**